stablehlo/reference/Element.cpp

```diff
@@ -100,11 +100,23 @@ Element map(const Element &lhs, const Element &rhs, IntegerFn integerFn,
                                      debugString(type).c_str()));
 }
 
-template <typename FloatFn, typename ComplexFn>
-Element mapWithUpcastToDouble(const Element &el, FloatFn floatFn,
+template <typename IntegerFn, typename BooleanFn, typename FloatFn,
+          typename ComplexFn>
+Element mapWithUpcastToDouble(const Element &el, IntegerFn integerFn,
+                              BooleanFn boolFn, FloatFn floatFn,
```

Comment:
"using update-to-double for non-float or on-complex does not seem right mainly because contrary to the function name we are not doing any upcasts". Yes I agree. It's not evident how is the "upcast to double" part supposed to work for e.g. boolean types. I think it would be better to implement support for division without using map.

Reply:
I needed the upcast function, but they do not return optional result instead throw errors and failures, so I could not use them as is.